lms/views/basic_lti_launch.py

```diff
@@ -135,10 +135,24 @@ def unconfigured_basic_lti_launch(self):
         we'll save it in our DB. Subsequent launches of the same assignment
         will then be DB-configured launches rather than unconfigured.
         """
+        oauth_consumer_key = self.request.lti_user.oauth_consumer_key
+
+        lms_url = self.request.params.get("custom_canvas_api_domain")
+
+        if not lms_url:
+            lms_url = self.request.find_service(name="ai_getter").lms_url(
+                oauth_consumer_key
+            )
+
         return {
             "content_item_return_url": self.request.route_url(
                 "module_item_configurations"
             ),
+            "google_client_id": self.request.registry.settings["google_client_id"],
+            "google_developer_key": self.request.registry.settings[
+                "google_developer_key"
+            ],
```

Comment:
As you can see it's a lot simpler if the backend code, templates and frontend code can all just assume that these settings exist. Less tests to write too